From ef0e890e77af1c86c744661d8d193f36f04925e9 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sat, 17 Feb 2024 02:01:49 +0100
Subject: dt: bindings: mfd: Add codec related properties to AC100 PMIC

This PMIC includes a co-packaged audio codec with analog and digital
parts. Add additional properties that describe power supplies to
the codec, and the node for analog part of the codec.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 .../bindings/mfd/x-powers,ac100.yaml          | 38 +++++++++++++++++++
 1 file changed, 38 insertions(+)

diff --git a/Documentation/devicetree/bindings/mfd/x-powers,ac100.yaml b/Documentation/devicetree/bindings/mfd/x-powers,ac100.yaml
index f3d8394b27e7..59ad898fbf85 100644
--- a/Documentation/devicetree/bindings/mfd/x-powers,ac100.yaml
+++ b/Documentation/devicetree/bindings/mfd/x-powers,ac100.yaml
@@ -34,11 +34,39 @@ properties:
         description: >
           Name of the 4M_adda clock exposed by the codec
 
+      "#sound-dai-cells":
+        const: 1
+
+      LDOIN-supply: true
+      AVCC-supply: true
+      VDDIO1-supply: true
+      VDDIO2-supply: true
+
     required:
       - "#clock-cells"
       - compatible
       - interrupts
       - clock-output-names
+      - "#sound-dai-cells"
+      - LDOIN-supply
+      - AVCC-supply
+      - VDDIO1-supply
+      - VDDIO2-supply
+
+    additionalProperties: false
+
+  codec-analog:
+    type: object
+
+    properties:
+      compatible:
+        const: x-powers,ac100-codec-analog
+
+      CPVDD-supply: true
+
+    required:
+      - compatible
+      - CPVDD-supply
 
     additionalProperties: false
 
@@ -94,12 +122,22 @@ examples:
             compatible = "x-powers,ac100";
             reg = <0xe89>;
 
+            ac100_codec_analog: codec-analog {
+                compatible = "x-powers,ac100-codec-analog";
+                CPVDD-supply = <&reg_aldo2>;
+            };
+
             ac100_codec: codec {
+                #sound-dai-cells = <1>;
                 compatible = "x-powers,ac100-codec";
                 interrupt-parent = <&r_pio>;
                 interrupts = <0 9 IRQ_TYPE_LEVEL_LOW>; /* PL9 */
                 #clock-cells = <0>;
                 clock-output-names = "4M_adda";
+                LDOIN-supply = <&reg_aldo2>;
+                AVCC-supply = <&reg_aldo3>;
+                VDDIO1-supply = <&reg_dcdc1>;
+                VDDIO2-supply = <&reg_dldo1>;
             };
 
             ac100_rtc: rtc {
-- 
2.35.3

